java-checks/src/main/java/org/sonar/java/checks/PreparedStatementAndResultSetCheck.java

```diff
@@ -86,7 +98,7 @@ protected void onMethodInvocationFound(MethodInvocationTree mit) {
         addIssue(mit, "PreparedStatement indices start at 1.");
       } else {
         Tree preparedStatementReference = getPreparedStatementReference(mit);
-        Integer numberParameters = getNumberParametersFromPreparedStatement(preparedStatementReference);
+        Integer numberParameters = getPreparedStatementNumberOfParameters(preparedStatementReference);
         if (numberParameters != null && methodFirstArgumentValue > numberParameters.intValue()) {
```

Comment:
@m-g-sonar unnecessary unboxing `.intValue()`. Doesn't eclipse notify you about it?